File: src/main/java/me/danetnaverno/editoni/blockstate/MinecraftLiquidState.java
package me.danetnaverno.editoni.blockstate;

import me.danetnaverno.editoni.world.BlockState;
import net.querz.nbt.tag.CompoundTag;

public class MinecraftLiquidState extends BlockState
{
    public final int level;

    public MinecraftLiquidState(CompoundTag properties)
    {
        super(properties);
        level = Integer.parseInt(properties.getString("level"));
    }
}

File: src/main/kotlin/me/danetnaverno/editoni/blockstate/MinecraftLiquidState.kt
package me.danetnaverno.editoni.blockstate

import net.querz.nbt.tag.CompoundTag

class MinecraftLiquidState(properties: CompoundTag) : BlockState(properties)
{
    val level = properties.getString("level").toInt() //properties.getInt("level") loves to throw an Exception for some reason
}